### DISCUSSION TITLE
Importing DummyOperator is not detected as an error

### DISCUSSION THREAD

**Discussion by esoyer** [apache/airflow#26631] [Q&A]
### Apache Airflow version

2.4.0

### What happened

After having upgraded our Airflow from 2.3.4 to 2.4.0, we observed that the DummyOperator was deleted (by the way, the release note does not talk about this at all). 

We ran our tests which check that the import_errors from our dag bag is an empty map, and the tests were passing even though the imports of DummyOperator were still there in the DAG python files. 

The test looks like something like this:

```
dagbag = DagBag(include_examples=False)
assert dagbag.import_errors == {}
```

### What you think should happen instead

We were expecting the test to fail since we are importing an Operator which does not exist anymore.

### How to reproduce

Upgrade to Airflow 2.4.0 and import the `DummyOperator` in one of your DAG python file. Run a test which checks that the `import_errors` from your Dag Bag containing your DAG with the wrong import is empty. It should pass even though the import is invalid.

### Operating System

MacOS Monterey Version 12.5

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by eladkal** [comment]
The `DummyOperator` was deprecated.
You can still use it.
see :
https://github.com/apache/airflow/blob/55d11464c047d2e74f34cdde75d90b633a231df2/airflow/operators/__init__.py#L51

We just replaced it  with PEP-562 dynamic attribute loading.

**Comment by potiuk** [comment]
BTW. It is mentioned in the release notes actually: https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-contrib-packages-and-deprecated-modules-are-dynamically-generated-26153-26179-26167


> The airflow.contrib packages and deprecated modules from Airflow 1.10 in airflow.hooks, airflow.operators, airflow.sensors packages are now dynamically generated modules and while users can continue using the deprecated contrib classes, they are no longer visible for static code check tools and will be reported as missing. It is recommended for the users to move to the non-deprecated classes.

**Reply by potiuk** [reply]
And BTW. you should switch to EmptyOperator @esoyer (It's as easy as search&replace in your DAGs) because it will be deleted sooner or later.

**Reply by esoyer** [reply]
Hi potiuk. We already did the search and replace that you mentioned which is indeed pretty simple, but we were simply caught off guard by this behavior which seemed strange to us. Thank you for your answer. ;)

**Reply by potiuk** [reply]
Well. We have no other way to communite those changes but by release notes, and about two year/ a yar of deprecation should be enough of a warning I think. 

BTW. Do you have any other ideas how we should communicate it so that it does not catach you off guards? What other means that long-time deprecation warning could be communicated better/differently (our end-goal is that you stop using those deprecated classes eventually)?
I'd love to hear if thare are any other ideas we could explore @esoyer - probably as a user who has been just 'caught' you can give us some ideas?
